### DISCUSSION TITLE
sidebar/sanfene/javase

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#82] [Announcements]
# sidebar/sanfene/javase

图文详解 53 道Java基础面试高频题，这次吊打面试官，我觉得稳了（手动 dog）

https://tobebetterjavaer.com/sidebar/sanfene/javase.html

**Comment by beautifulsakuragrass** [comment]
王哥，你这份资料思路很不错，学习了。上面有一个小小瑕疵，就是有一张图有误：《Java数据类型》 这个图需要修改一下。特意用github账号登陆了。

**Comment by beautifulsakuragrass** [comment]
这个资料很不错，我也来帮忙维护。

**Reply by itwanger** [reply]
期待哦，issue 和 PR 都可以

**Comment by caoshengshi** [comment]
44、BIO、NIO、AIO图片有误

**Reply by ganmin** [reply]
我也看到了

**Comment by ganmin** [comment]
49.反射里的第二句话“
而如果，我们想在时候动态地获取类信息、创建类实例、调用类方法这时候就要用到反射”。。少了“运行”二字

**Comment by condy123** [comment]
2 << 3   你打死我，我也说它等于16

**Comment by LVXUEYANGTiger** [comment]
看完了

**Comment by condy123** [comment]
我是唐琼，谢谢你的来信....2013祝你幸福...

**Comment by Durantttttttttttttt** [comment]
打卡

**Comment by Eivae** [comment]
float的结构图那里是不是画错了？float的指数（或者说阶码部分）不是才8位吗？怎么变成10位了？

**Comment by condy123** [comment]
我是唐琼，谢谢你的来信....2013祝你幸福...

**Comment by condy123** [comment]
我是唐琼，谢谢你的来信....2013祝你幸福...

**Comment by JingjieMa2333** [comment]
15.float 是怎么表示小数的这部分有个小瑕疵，25.125的二进制科学计数法是1.10010001×2^4。因为在 IEEE 754 浮点数标准中，采用了 隐藏位（隐含的 1） 的机制来优化存储空间，所以尾数是10010010000000000000000，而不是简单的转换为二进制科学计数法1.001001×2^4

**Comment by condy123** [comment]
我是唐琼，谢谢你的来信....2013祝你幸福...

**Comment by strings32767** [comment]
15.float表示方式 描述有误 在IEEE754浮点数标准中，指数部分（阶码）应该占8位，尾数部分应该占23位。

**Comment by condy123** [comment]
我是唐琼，谢谢你的来信....2013祝你幸福...
